### PR TITLE
femtolisp: init at 2023-07-12

### PR DESCRIPTION
**Pull request by majiru**
###### Description of changes

A compact interpreter for a minimal lisp/scheme dialect.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/femtolisp/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+, unstableGitUpdater
+}:
+
+stdenv.mkDerivation rec {
+  pname = "femtolisp";
+  version = "unstable-2023-07-12";
+
+  src = fetchFromSourcehut {
+    owner = "~ft";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/femtolisp/default.nix:13]:
```suggestion
    repo = "femtolisp";
```

File: pkgs/development/interpreters/femtolisp/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    rev = "b3a21a0ff408e559639f6c31e1a2ab970787567f";
+    sha256 = "sha256-PE/xYhfhn0xv/kJWsS07fOF2n5sXP666vy7OVaNxc7Y=";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/femtolisp/default.nix:15]:
```suggestion
    hash = "sha256-PE/xYhfhn0xv/kJWsS07fOF2n5sXP666vy7OVaNxc7Y=";
```

File: pkgs/development/interpreters/femtolisp/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  strictDeps = true;
+  enableParallelBuilding = true;
+  passthru.updateScript = unstableGitUpdater { };
+  installPhase = ''
+    install -Dm755 -t $out/bin/ flisp
+  '';
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/femtolisp/default.nix:23]:
```suggestion
  strictDeps = true;
  
  enableParallelBuilding = true;
  
  installPhase = ''
    runHook preInstall
    
    install -Dm755 -t $out/bin/ flisp
    
    runHook postInstall
  '';

  passthru.updateScript = unstableGitUpdater { };
```

File: pkgs/development/interpreters/femtolisp/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  meta = with lib; {
+    description = "A compact interpreter for a minimal lisp/scheme dialect.";
+    homepage = "https://git.sr.ht/~ft/femtolisp";
+    license = with licenses; [ mit bsd3 ];
+    maintainers = with maintainers; [ moody ];
+    platforms = platforms.linux;
+  };
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/femtolisp/default.nix:31]:
```suggestion
  meta = {
    description = "A compact interpreter for a minimal lisp/scheme dialect.";
    homepage = "https://git.sr.ht/~ft/femtolisp";
    license = with lib.licenses; [ mit bsd3 ];
    maintainers = with lib.maintainers; [ moody ];
    platforms = lib.platforms.unix;
  };
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/interpreters/femtolisp/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    install -Dm755 -t $out/bin/ flisp
+
+    runHook postInstall
+  '';
+
+  passthru.updateScript = unstableGitUpdater { };
+
+  meta = {
+    description = "A compact interpreter for a minimal lisp/scheme dialect.";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/femtolisp/default.nix:33]:
```suggestion
    description = "A compact interpreter for a minimal lisp/scheme dialect";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Besides, lgtm

**Comment by majiru** [comment]
> Besides, lgtm

Thank you! I did have to change the meta.platforms back to just linux, there is currently an issue with getting it to compile on darwin related to endian.h include path shenanigans.

**Comment by AndersonTorres** [comment]
> > Besides, lgtm
> 
> Thank you! I did have to change the meta.platforms back to just linux, there is currently an issue with getting it to compile on darwin related to endian.h include path shenanigans.

In this case, you can set `broken = stdenv.isDarwin;` inside meta.

(Well, supposing the femtolisp developers wanted it to run on Darwin...)

**Pull request merged**